trin-core/src/portalnet/find/iterators/findcontent.rs

```diff
@@ -0,0 +1,715 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+// This basis of this file has been taken from the rust-libp2p codebase:
+// https://github.com/libp2p/rust-libp2p
+
+use std::collections::btree_map::{BTreeMap, Entry};
+use std::time::Instant;
+
+use discv5::kbucket::{Distance, Key};
+
+use super::super::query_pool::QueryState;
+use super::query::{Query, QueryConfig, QueryPeer, QueryPeerState, QueryProgress};
+
+pub enum FindContentResponse<TNodeId> {
```

Comment:
*nit* Iiuc, this name is slightly confusing. Technically this represents a `Content` response to a `FindContent` query right? I guess `FindContentQueryResponse` is more explicit but wordy - but if this struct is limited to this module it makes more sense within this context. Do these structs (& `FindContentQueryResult` need to be `pub`)?

Reply:
Yes, this represents a `Content` response. Yes, these structs need to be `pub`, because those types will be exposed/used on calls to `into_result` and `on_success`.

I don't mind verbosity if it makes things easier to understand. The reason that we create a new type here is that `Content` also contains a `ConnectionId` variant, which we will not use here.